src/utils/rest-api.js

```diff
@@ -1,95 +1,134 @@
+import {store} from '../redux/store'
 /**
  * Copyright (c) 2020, RTE (http://www.rte-france.com)
  * This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
+let PREFIX_CASE_QUERIES;
+let PREFIX_STUDY_QUERIES;
+if (process.env.REACT_APP_USE_AUTHENTICATION === "true") {
+    PREFIX_CASE_QUERIES = process.env.REACT_APP_API_GATEWAY + "/case";
+    PREFIX_STUDY_QUERIES = process.env.REACT_APP_API_GATEWAY + "/study";
+} else {
+    PREFIX_CASE_QUERIES = process.env.REACT_APP_API_CASE_SERVER;
+    PREFIX_STUDY_QUERIES = process.env.REACT_APP_API_STUDY_SERVER;
+}
+
+function getHeaders() {
+    const state = store.getState();
+    const token = state.user.id_token;
+    return new Headers({
+        'Authorization': 'Bearer '+ token,
+    });
+}
+
 export function fetchStudies() {
     console.info("Fetching studies...");
-    const fetchStudiesUrl = process.env.REACT_APP_API_STUDY_SERVER + "/v1/studies";
+    const fetchStudiesUrl = PREFIX_STUDY_QUERIES + "/v1/studies";
     console.debug(fetchStudiesUrl);
-    return fetch(fetchStudiesUrl)
-        .then(response => response.json());
+    return fetch(fetchStudiesUrl, {
```

Comment:
create a method backendFetch(..) which delegates to fetchs and puts the auth header when USE_AUTHENTICATION is true